src/main/java/cd/go/plugin/config/yaml/YamlConfigPlugin.java

```diff
@@ -114,6 +116,24 @@ private void ensureConfigured() {
         }
     }
 
+    private GoPluginApiResponse handleGetConfigFiles(GoPluginApiRequest request) {
+        return handlingErrors(() -> {
+            ParsedRequest parsed = ParsedRequest.parse(request);
+            File baseDir = new File(parsed.getStringParam("directory"));
+            String pattern = parsed.getConfigurationKey(PLUGIN_SETTINGS_FILE_PATTERN);
+
+            if (isBlank(pattern)) {
+                pattern = getFilePattern();
+            }
+
+            String[] files = new AntDirectoryScanner().getFilesMatchingPattern(baseDir, pattern);
```

Comment:
This whole block 
```
            ParsedRequest parsed = ParsedRequest.parse(request);
            File baseDir = new File(parsed.getStringParam("directory"));
            String pattern = parsed.getConfigurationKey(PLUGIN_SETTINGS_FILE_PATTERN);

            if (isBlank(pattern)) {
                pattern = getFilePattern();
            }

            YamlConfigParser parser = new YamlConfigParser();

            String[] files = new AntDirectoryScanner().getFilesMatchingPattern(baseDir, pattern);
```
is almost a copy-paste from `handleParseDirectoryRequest`.

Can you refactor it so that there is a function which returns matches files. Then use that function in both `handleGetConfigFiles` and `handleParseDirectoryRequest`?
This way we can be certain that what plugin uses in actual parsing and in listing the files is consistent

Reply:
How does what I pushed look?